validator/src/micro.rs

```diff
@@ -95,26 +88,36 @@ impl<TValidatorNetwork: ValidatorNetwork + 'static> NextProduceMicroBlockEvent<T
                 if !in_current_state(&blockchain.head()) {
                     break Some(None);
                 } else if self.is_our_turn(&blockchain) {
-                    if !logged {
+                    // We want to produce a block at the expected timestamp for this block in this epoch
+                    // as it is calculated by the reward function
+                    let last_election_block = blockchain.election_head();
+                    let target_block_ts = last_election_block.header.timestamp
+                        + self.block_separation_time.as_secs()
+                            * 1000
```

Comment:
You can do this to avoid the `*1000` line.
```suggestion
                        + self.block_separation_time.as_millis() as u64
```